MSVC warning in 64bit following xxHash update.

After the update of xxHash in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4633 the roll of VVLs in Dawn and ANGLE is failing because of an MSVC warning being triggered. Looking at the code I don't really understand what the problem is so I'll add a suppression in the BUILD.gn file. Out of curiosity, @kaadam, do you think that https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4639 would fix this issue (seems unlikely, but we never know).

The exact warning emitted is:
```
ninja -t msvc -e environment.x64 -- "C:\b\s\w\ir\cache\win_toolchain\vs_files\1023ce2e82\VC\Tools\MSVC\14.29.30133\bin\Hostx64\x64\cl.exe" /c ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/xxhash.c /nologo /showIncludes -DUSE_AURA=1 -D_HAS_EXCEPTIONS=0 -D__STD_C -D_CRT_RAND_S -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_DEPRECATE -D_ATL_NO_OPENGL -D_WINDOWS -DCERT_CHAIN_PARA_HAS_EXTRA_FIELDS -DPSAPI_VERSION=2 -DWIN32 -D_SECURE_ATL -DWINAPI_FAMILY=WINAPI_FAMILY_DESKTOP_APP -DWIN32_LEAN_AND_MEAN -DNOMINMAX -DNTDDI_VERSION=NTDDI_WIN10_FE -D_WIN32_WINNT=0x0A00 -DWINVER=0x0A00 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -D_CRT_NONSTDC_NO_WARNINGS -D_WINSOCK_DEPRECATED_NO_WARNINGS -DVULKAN_NON_CMAKE_BUILD -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_WIN32_KHR -I../../third_party/vulkan-deps/vulkan-validation-layers/src/layers -I../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/generated -I../../third_party/vulkan-deps/spirv-headers/src/include -I../.. -Igen -I../../third_party/vulkan-deps/vulkan-headers/src/include /Gy /FS /bigobj /utf-8 /Zc:sizedDealloc- /wd4117 /D__DATE__= /D__TIME__= /D__TIMESTAMP__= /O1 /Ob2 /Oy- /Zc:inline /Gw /Oi /Zi /MT /wd4800 /wd4267 /WX /wd4244 /std:c11 /TC /Foobj/third_party/vulkan-deps/vulkan-validation-layers/src/vulkan_layer_utils/xxhash.obj /Fd"obj/third_party/vulkan-deps/vulkan-validation-layers/src/vulkan_layer_utils_c.pdb"
C:\b\s\w\ir\cache\builder\dawn\third_party\vulkan-deps\vulkan-validation-layers\src\layers\xxhash.h(4578): error C2220: the following warning is treated as an error
C:\b\s\w\ir\cache\builder\dawn\third_party\vulkan-deps\vulkan-validation-layers\src\layers\xxhash.h(4578): warning C4028: formal parameter 4 different from declaration
```

--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -58,6 +58,7 @@ if(BUILD_LAYER_SUPPORT_FILES)
         vk_layer_logging.h
         vk_layer_utils.h
         vk_layer_utils.cpp
+        vk_xxhash.h
         xxhash.h
         xxhash.c
         generated/vk_format_utils.h
@@ -202,7 +203,9 @@ set(CORE_VALIDATION_LIBRARY_FILES
     generated/synchronization_validation_types.cpp
     gpu_validation.cpp
     generated/corechecks_optick_instrumentation.cpp
-    xxhash.c)
+    xxhash.c
+    xxhash.h
+    vk_xxhash.h)
 
 set(OBJECT_LIFETIMES_LIBRARY_FILES
     generated/object_tracker.cpp

--- a/layers/layer_options.cpp
+++ b/layers/layer_options.cpp
@@ -247,9 +247,9 @@ void CreateFilterMessageIdList(std::string raw_id_list, std::string delimiter, s
         token = GetNextToken(&raw_id_list, delimiter, &pos);
         uint32_t int_id = TokenToUint(token);
         if (int_id == 0) {
-            size_t id_hash = XXH32(token.c_str(), strlen(token.c_str()), 8);  // String
+            const uint32_t id_hash = XXH32(token.data(), token.size(), 8);  // String
             if (id_hash != 0) {
-                int_id = static_cast<uint32_t>(id_hash);
+                int_id = id_hash;
             }
         }
         if ((int_id != 0) && (std::find(filter_list.begin(), filter_list.end(), int_id)) == filter_list.end()) {

--- a/layers/shader_validation.cpp
+++ b/layers/shader_validation.cpp
@@ -37,8 +37,7 @@
 #include "chassis.h"
 #include "core_validation.h"
 #include "spirv_grammar_helper.h"
-
-#include "xxhash.h"
+#include "vk_xxhash.h"
 
 static shader_stage_attributes shader_stage_attribs[] = {
     {"vertex shader", false, false, VK_SHADER_STAGE_VERTEX_BIT},

--- a/layers/vk_layer_logging.h
+++ b/layers/vk_layer_logging.h
@@ -55,7 +55,7 @@
 #include "vk_validation_error_messages.h"
 #include "vk_layer_dispatch_table.h"
 #include "vk_safe_struct.h"
-#include "xxhash.h"
+#include "vk_xxhash.h"
 
 // Suppress unused warning on Linux
 #if defined(__GNUC__)
@@ -671,7 +671,7 @@ static inline bool LogMsgEnabled(const debug_report_data *debug_data, const std:
         return false;
     }
     // If message is in filter list, bail out very early
-    uint32_t message_id = XXH32(vuid_text.c_str(), strlen(vuid_text.c_str()), 8);
+    const uint32_t message_id = XXH32(vuid_text.data(), vuid_text.size(), 8);
     if (std::find(debug_data->filter_message_ids.begin(), debug_data->filter_message_ids.end(), message_id)
         != debug_data->filter_message_ids.end()) {
         return false;

--- a/layers/vk_xxhash.h
+++ b/layers/vk_xxhash.h
@@ -1,0 +1,30 @@
+/* Copyright (c) 2015-2022 The Khronos Group Inc.
+ * Copyright (c) 2015-2022 Valve Corporation
+ * Copyright (c) 2015-2022 LunarG, Inc.
+ * Copyright (C) 2015-2022 Google Inc.
+ * Modifications Copyright (C) 2020 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Juan Ramos
+ */
+
+// XXH_NO_LONG_LONG: removes compilation of algorithms relying on 64-bit types (XXH3 and XXH64). Only XXH32 will be compiled.
+// We only need XXH32 due to restrictions requiring a 32 bit hash. This also reduces binary size.
+//
+// v0.8.1 also has compilation issues that are removed by setting this define.
+// https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4639
+// https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4640
+#define XXH_NO_LONG_LONG
+
+#include "xxhash.h"

